drivers/sensor/bmi08x/bmi08x_gyro.c

```diff
@@ -0,0 +1,472 @@
+ * or INT2
+ */
+#if defined(CONFIG_BMI08X_GYRO_TRIGGER) || BMI08X_GYRO_ANY_INST_HAS_DATA_SYNC
+	/* set gyro ints */
+	/* set ints */
+	if (bmi08x_gyro_byte_write(dev, BMI08X_REG_GYRO_INT_CTRL, 0x80) < 0) {
+		LOG_ERR("Failed to map interrupts.");
+		return -EIO;
+	}
+	if (bmi08x_gyro_byte_write(dev, BMI08X_REG_GYRO_INT3_INT4_IO_CONF, config->int3_4_conf_io) <
+	    0) {
+		LOG_ERR("Failed to map interrupts.");
+		return -EIO;
+	}
+	if (bmi08x_gyro_byte_write(dev, BMI08X_REG_GYRO_INT3_INT4_IO_MAP, config->int3_4_map) < 0) {
+		LOG_ERR("Failed to map interrupts.");
+		return -EIO;
+	}
+#endif
+
+	return 0;
+}
+
+#define BMI08X_CONFIG_SPI(inst)                                                                    \
+	.bus.spi = SPI_DT_SPEC_INST_GET(                                                           \
+		inst, SPI_OP_MODE_MASTER | SPI_TRANSFER_MSB | SPI_WORD_SET(8), 2),
+
+#define BMI08X_CONFIG_I2C(inst) .bus.i2c = I2C_DT_SPEC_INST_GET(inst),
+
+#define BMI08X_GYRO_TRIG(inst)                                                                     \
+	.int3_4_map = DT_INST_PROP(inst, int3_4_map_io),                                           \
+	.int3_4_conf_io = DT_INST_PROP(inst, int3_4_conf_io),
+
+#if BMI08X_GYRO_ANY_INST_HAS_DATA_SYNC
+/* the bmi08x-gyro should not have trigger mode with data-sync enabled */
+BUILD_ASSERT(CONFIG_BMI08X_GYRO_TRIGGER_NONE,
+	     "Only none trigger type allowed for bmi08x-gyro with data-sync enabled");
+/* with data-sync, one of the int pins should be wired directory to the accel's int pins, their
+ * config should be defined
+ */
+#define BMI08X_GYRO_TRIGGER_PINS(inst) BMI08X_GYRO_TRIG(inst)
+#else
+#define BMI08X_GYRO_TRIGGER_PINS(inst)                                                             \
+	IF_ENABLED(CONFIG_BMI08X_GYRO_TRIGGER, (BMI08X_GYRO_TRIG(inst)))
+#endif
+
+#define BMI08X_CREATE_INST(inst)                                                                   \
+                                                                                                   \
+	static struct bmi08x_gyro_data bmi08x_drv_##inst;                                          \
+                                                                                                   \
+	static const struct bmi08x_gyro_config bmi08x_config_##inst = {                            \
+		COND_CODE_1(DT_INST_ON_BUS(inst, spi), (BMI08X_CONFIG_SPI(inst)),                  \
+			    (BMI08X_CONFIG_I2C(inst)))                                             \
+			.api = COND_CODE_1(DT_INST_ON_BUS(inst, spi), (&bmi08x_spi_api),           \
+					   (&bmi08x_i2c_api)),                                     \
+		IF_ENABLED(CONFIG_BMI08X_GYRO_TRIGGER,                                             \
+			   (.int_gpio = GPIO_DT_SPEC_INST_GET(inst, int_gpios),))                 \
+			.gyro_hz = DT_INST_ENUM_IDX(inst, gyro_hz),                                \
+		BMI08X_GYRO_TRIGGER_PINS(inst).gyro_fs = DT_INST_ENUM_IDX(inst, gyro_fs),          \
```

Comment:
`gyro_fs = DT_INST_ENUM_IDX(inst, gyro_fs)`
is a bug, you need change it to `.gyro_fs = DT_INST_PROP(inst, gyro_fs),`
get range from dts, not the index.

Reply:
Nope, the enum idx is to correspond with the reg value, please take a look at the datasheet again

Reply:
the function `bmi08x_range_to_reg_val` need range value not index, and not the register value.